execute/plugin.go

```diff
@@ -209,50 +210,55 @@ func getPendingReportsForExecution(
 				"skippedCommitRoots", skippedCommitRoots,
 			)
 		}
-		reports = filtered
-
-		lggr.Debugw("grouped reports", "selector", selector, "reports", reports, "count", len(reports))
-		sort.Slice(reports, func(i, j int) bool {
-			return reports[i].SequenceNumberRange.Start() < reports[j].SequenceNumberRange.Start()
+		filteredReports[selector] = filtered
+
+		lggr.Debugw("grouped reports",
+			"selector", selector,
+			"reports", filteredReports[selector],
+			"count", len(filteredReports[selector]))
+		sort.Slice(filteredReports[selector], func(i, j int) bool {
+			return filteredReports[selector][i].SequenceNumberRange.Start() <
+				filteredReports[selector][j].SequenceNumberRange.Start()
 		})
 		// todo: remove this logs after investigating whether the sorting above can be safely removed
-		lggr.Debugw("sorted reports", "selector", selector, "reports", reports, "count", len(reports))
+		lggr.Debugw("sorted reports",
+			"selector", selector,
+			"reports", filteredReports[selector],
+			"count", len(filteredReports[selector]))
```

Comment:
I think we can simplify the logging here and just keep the sort, better to be safe than sorry. Also I think its cleaner to do the sort prior to the map assignment? And `filtered` as a name is not super expressive, maybe change to `executableReports`?

```golang
// ensure we are always executing from earliest to latest report (seq nr order).
sort.Slice(executableReports, func(i, j int) bool {
    return executableReports[i].SequenceNumberRange.Start() <
        executableReports[j].SequenceNumberRange.Start()
})
executableReportsPerChain[selector] = executableReports
lggr.Debugw("grouped reports and sorted",
    "selector", selector,
    "reports", executableReportsPerChain[selector],
    "count", len(executableReportsPerChain[selector]))
```